src/FormTaxonomies/Actions/UpdateFormTaxonomies.php

```diff
@@ -0,0 +1,25 @@
+<?php
+
+namespace Give\FormTaxonomies\Actions;
+
+use Give\DonationForms\Models\DonationForm;
+
+/**
+ * @unreleased
+ */
+class UpdateFormTaxonomies
+{
+    /**
+     * @unreleased
+     */
+    public function __invoke(DonationForm $form)
+    {
+        if($form->settings->formTags) {
+            wp_set_post_terms($form->id, array_column($form->settings->formTags, 'id'), 'give_forms_tag');
+        }
+
+        if($form->settings->formCategories) {
+            wp_set_post_terms($form->id, array_column($form->settings->formCategories, 'id'), 'give_forms_category');
+        }
```

Comment:
If we decide to go with notion of taxonomies living outside the concept of donation forms and the form settings should not store their values then this would the alternative approach (this is how i'm currently recommending third parties create their own form settings so we are not storing their values in our form settings object.):

```
add_action('givewp_form_builder_updated', function($form) {
    if (isset($_POST['settings'])) {
        $settings = json_decode(give_clean($_POST['settings']), true);
        if (isset($settings['formTags'])) {
            $formTags = give_clean($settings['formTags']);

            wp_set_post_terms($form->id, array_column($formTags, 'id'), 'give_forms_tag');
        }
    }
});
```

Reply:
Ok, that makes sense. I had considered reaching into the request directly, but it "felt hack-y", but we don't have an official way to pass "extra" data to the server on publish.

Reply:
Yeah I agree on the hacky lol, I could see adding a couple additional params to these form builder actions to pass the request along.  When it comes to 3rd parties adding their own settings, I really just want them to handle the storage themselves so we are not persisting unknown data.

Reply:
Resolved, see [558e723](https://github.com/impress-org/givewp/pull/7463/commits/558e723908fa1e8975fd3bc511b5b0dae76e2a47).